src/main/java/world/bentobox/caveblock/generators/populators/MaterialPopulator.java

```diff
@@ -62,99 +62,82 @@ public MaterialPopulator(CaveBlock addon)
-                                case 3:
-                                    x = Math.max(0, x - 1);
-                                    break;
-                                case 4:
-                                    y = Math.max(1, y - 1);
-                                    break;
-                                case 5:
-                                    z = Math.max(0, z - 1);
-                                    break;
-                                }
-
-                                block = chunk.getBlock(x, y, z);
-
-                                continuePlacing = this.isValidBlock(world, block, x, z) &&
-                                        packSize > 0 &&
-                                        (block.getType().equals(chances.get(world.getEnvironment()).mainMaterial) ||
-                                                block.getType().equals(entry.getKey()));
+                            case 0:
+                                x = Math.min(15, x + 1);
+                                break;
+                            case 1:
+                                y = Math.min(worldHeight - 2, y + 1);
+                                break;
+                            case 2:
+                                z = Math.min(15, z + 1);
+                                break;
+                            case 3:
+                                x = Math.max(0, x - 1);
+                                break;
+                            case 4:
+                                y = Math.max(1, y - 1);
+                                break;
+                            case 5:
+                                z = Math.max(0, z - 1);
+                                break;
                             }
+
+                            block = chunk.getBlock(x, y, z);
+
+                            continuePlacing = packSize > 0 &&
+                                    (block.getType().equals(chances.get(world.getEnvironment()).mainMaterial) ||
+                                            block.getType().equals(entry.getKey()));
                         }
                     }
                 }
             }
         }
     }
 
-
-    /**
-     * This method checks if all chunks around given block is generated.
-     * @param world World in which block is located
-     * @param block Block that must be checked.
-     * @param x Block x-index in chunk
-     * @param z Block z-index in chunk
-     * @return true, if all chunks around given block are generated.
-     */
-    private boolean isValidBlock(World world, Block block, int x, int z)
-    {
-        return x > 0 && x < 15 && z > 0 && z < 15 ||
-                world.isChunkGenerated(block.getX() + 1, block.getZ()) &&
-                world.isChunkGenerated(block.getX() - 1, block.getZ()) &&
-                world.isChunkGenerated(block.getX(), block.getZ() - 1) &&
-                world.isChunkGenerated(block.getX(), block.getZ() + 1);
-    }
```

Comment:
If the original coordinates are not on the edge of the chunk then this code should not be needed. Apparently vanilla never mods any blocks on chunk boundaries for this reason.

Reply:
world.isChunkGenerated() allowed to do it, if chunk near is generated.